2023/pav_U1/main.cpp

```diff
@@ -0,0 +1,204 @@
+#include <fstream>
+#include <iostream>
+#include <vector>
+#include <sstream>
+using namespace std;
+
+struct PradiniaiDuomenys
+{
+    string pavadinimasDavusios;
+    string pavadinimasGavusios;
+    int taskuKiekis = 0;
+};
+
+struct Salis
+{
+    string pavadinimas;
+    int taskai = 0;
+    int kiekKartuGavo12T = 0;
+};
+
+void skaitymas(vector<PradiniaiDuomenys> &duomenys)
+{
+    ifstream data("Eurovizija.csv");
+    PradiniaiDuomenys tempDuomenys;
+    string line;
+
+    while (getline(data, line))
+    {
+        stringstream eilute(line);
+        string saliesPavadinimas;
+
+        getline(eilute, saliesPavadinimas, ';');
+        tempDuomenys.pavadinimasDavusios = saliesPavadinimas;
+
+        getline(eilute, saliesPavadinimas, ';');
+        tempDuomenys.pavadinimasGavusios = saliesPavadinimas;
+
+        getline(eilute, saliesPavadinimas, ';');
+        tempDuomenys.taskuKiekis = stoi(saliesPavadinimas);
+        duomenys.push_back(tempDuomenys);
+    }
```

Comment:
Perfect choice, good googling! Noice! In the exam, you can forget about those sophisticated methods, hence simple while or for loops can be used to read CSV files, one example:
```
while (getline(data, line))
    {
        const char DELIMITER = ';';
        int column = 0;
        string actingString = "";
        
        for (int i = 0; i < line.size(); i++) {
            
            if (line[i] == DELIMITER && column == 0) {
                tempDuomenys.pavadinimasDavusios = actingString;
                actingString = "";
                column++;
                continue;
            }
            
            if (line[i] == DELIMITER && column == 1) {
                tempDuomenys.pavadinimasGavusios = actingString;
                actingString = "";
                column++;
                continue;
            }
            
            actingString += line[i];
        }
        tempDuomenys.taskuKiekis = stoi(actingString);

        duomenys.push_back(tempDuomenys);
    }
    data.close();
```

Reply:
The most important thing is not to forget or know how to find documentation and use of getline method.

Reply:
Another way:
```
while (getline(data, line))
    {
        const char DELIMITER = ';';
        string actingString = "";
        int i = 0;

        while (line[i] != DELIMITER) {
            actingString += line[i++];
        }
        tempDuomenys.pavadinimasDavusios = actingString;
        actingString = "";
        i++;

        while (line[i] != DELIMITER) {
            actingString += line[i++];
        }
        tempDuomenys.pavadinimasGavusios = actingString;
        actingString = "";
        i++;

        while (i < line.size()) {
            actingString += line[i++];
        }
        tempDuomenys.taskuKiekis = stoi(actingString);

        duomenys.push_back(tempDuomenys);
    }
    data.close();
```